### DISCUSSION TITLE
When find path or subgraph will support vertex property filtering?

### DISCUSSION THREAD

**Discussion by imrefazekas** [vesoft-inc/nebula#5752] [Ideas]
Dear Team,

We are storing financial data and it would be crucial for us to support path finding filtering out given verteces by properties.
For example: economic connection between entities filtering out the "service provider" way so shopping centres and online shops? That would connect everyone to everyone... So that would filter out the noise, 95% of the possible paths in our case...
And being a service provider is the property of an vertex (legal entities) and not an edge for sure ...
And timeline you might can give on this?

**Comment by wey-gu** [comment]
Dear @imrefazekas the secret of FIND PATH/GET SUBGRAPH to being more efficient than `MATCH` was it doesn't read data from vertices' prop during the BFS for now.

*update: get subgraph support filtering on target vertex prop.*

cc @MuYiYong , it seems vertex prop filtering is indeed helpful in such a use case by @imrefazekas .

btw. by any chance we could move some of the info from vertices to edges in your case to enable the pruning of BFS in FIND PATH/GET SUBGRAPH?

**Reply by imrefazekas** [reply]
Unfortunately it is not possible to move to the edge because there are multiple edges between nodes, companies and a company can be "service provider" but an edge can not.

Should I use GO statement instead? GO can be used to get the nodes where I can filter by node attributes ...

**Reply by wey-gu** [reply]
GO was designed to only return the last hop info. except we explicitly write GO step after step connect with the pipeline and pass info per each step. This comes with limitations(fixed hops, too) but due to its explicit expression, we could have the best performance(compared to MATCH).

```
# an example w/o WHERE clause
GO 1 STEPS FROM 'A1', 'A2' OVER edge1 
YIELD DISTINCT dst(edge) as B, src(edge) as A | 
GO 1 STEPS FROM $-.B OVER edge2 
YIELD DISTINCT dst(edge) as C, $-.A as A, $-.B as B | 
GO 1 STEPS FROM $-.C OVER edge3 
YIELD DISTINCT dst(edge) as D, $-.A as A, $-.B as B, $-.C as C | 
GO 1 STEPS FROM $-.D OVER edge4 
YIELD DISTINCT $-.A as A, $-.B as B, $-.C as C, $-.D as D, dst(edge) as E
```
